### PR TITLE
Add unit tests for ifplugin/data_resync

### PR DESCRIPTION
**Pull request by deathbeam**
Signed-off-by: Tomas Slusny <tslusny@cisco.com>

### CODE REVIEW

File: plugins/vpp/ifplugin/data_resync_test.go [hunk 1]
```diff
@@ -0,0 +1,1540 @@
...
+			Message: &interfaces.SwInterfaceDetails{
+				InterfaceName: []byte("memif0"),
+				Tag:           []byte("test2"),
+				AdminUpDown:   1,
+				LinkMtu:       9216, // Default MTU
+				SwIfIndex:     1,
+			},
+		},
+		{
+			Id:      1034,
```
Comment (ondrej-fabry) [inline plugins/vpp/ifplugin/data_resync_test.go]: You could compact these like this to save some lines:

```go
...
{Id: 1034, Ping: false, Message: ...},
...
```

File: plugins/vpp/ifplugin/data_resync.go [hunk 2]
```diff
@@ -464,6 +464,7 @@ func (plugin *NatConfigurator) ResyncDNat(nbDNatConfig []*nat.Nat44DNat_DNatConf
 	// Correlate with existing config
 	for _, nbDNat := range nbDNatConfig {
 		for _, vppDNat := range vppDNatCfg.DnatConfigs {
+			plugin.log.Infof("nbdnatlabel %v , vpp %v", nbDNat.Label, vppDNat.Label)
```
Comment (ondrej-fabry) [inline plugins/vpp/ifplugin/data_resync.go]: Should probably be `Debugf`.

Comment (deathbeam) [reply plugins/vpp/ifplugin/data_resync.go]: Oh, this should not be in this diff at all, oops

File: plugins/vpp/ifplugin/data_resync_test.go [hunk 3]
```diff
@@ -0,0 +1,1334 @@
...
+				SocketFilename: "test",
+				Master:         true,
+				Id:             1,
+			},
+		},
+	}
+
+	errs := plugin.Resync(intfaces)
+	Expect(errs).To(BeEmpty())
+}
```
[the remark this answers is not in the document]
Comment (deathbeam) [reply plugins/vpp/ifplugin/data_resync_test.go]: Updated

**Pull request merged**